api/scanner/media_encoding/executable_worker/executable_worker.go

```diff
@@ -1,163 +1,50 @@
 package executable_worker
 
 import (
-	"fmt"
-	"io/ioutil"
 	"log"
-	"os"
-	"os/exec"
-	"strings"
 
 	"github.com/kkovaletp/photoview/api/utils"
-	"github.com/pkg/errors"
 	"gopkg.in/vansante/go-ffprobe.v2"
 )
 
 func InitializeExecutableWorkers() {
+	var err error
 	DarktableCli = newDarktableWorker()
-	FfmpegCli = newFfmpegWorker()
+	FfmpegCli, err = NewFfmpegWorker()
+	if err != nil {
+		log.Printf("An error occured when trying to initialize FFmpeg worker:\n%s", err)
+	}
 }
 
 var DarktableCli *DarktableWorker = nil
 var FfmpegCli *FfmpegWorker = nil
 
 type ExecutableWorker interface {
 	Path() string
-}
-
-type DarktableWorker struct {
-	path string
-}
-
-type FfmpegWorker struct {
-	path string
+	IsInstalled() bool
+	EncodeMp4(inputPath string, outputPath string) error
+	EncodeVideoThumbnail(inputPath string, outputPath string, probeData *ffprobe.ProbeData) error
```

Comment:
I don't understand why a worker interface contains videos method, but not `EncodeJpeg()` method.

Reply:
I have no idea) 
Originally it contained only the `Path()`. Then at some point of time GPT added everything else, it contains now. And I don't see an impact, as the Darktable is still called for image processing tasks...
If you think that it should be there (and could shortly explain what is the impact in practice), I can add it there.

Reply:
I suggest removing this interface. I didn't find anywhere using it.